### ISSUE TITLE
setTransportGate has no effect

### ISSUE THREAD

**Issue by wangbax** [getsentry/sentry-java#2565]
### Integration

sentry-android

### Build System

Gradle

### AGP Version

7.0.2

### Proguard

Enabled

### Version

6.14.0

### Steps to Reproduce

1. init sentry
2. reset`options.setTransportGate`
```kotlin
SentryAndroid.init(context) { options ->
    options.setTransportGate {
        true
    }
}
```

### Expected Result

Network status can be customized

### Actual Result

Built-in settings used.

<img width="753" alt="image" src="https://user-images.githubusercontent.com/5964735/221127387-0c62fd4e-343a-40f3-8fb3-f99e406035bc.png">

<img width="787" alt="image" src="https://user-images.githubusercontent.com/5964735/221127327-b950cbc6-1250-4605-85e9-021c799574ba.png">

**Comment by wangbax** [comment]
I found some relevant logic [SentryClient#init](https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/SentryClient.java#L55) & [AsyncHttpTransportFactory#create](https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/AsyncHttpTransportFactory.java#L21)

**Comment by romtsn** [comment]
Yeah, I can see why this is happening, we'll make sure to provide a fix, thanks for reporting!

**Comment by markushi** [comment]
Let's ensure other options are not overwritten after the user config lambda was executed.
